src/streaming-ingestion/redshift/streaming-ingestion-associate-role.ts

```diff
@@ -0,0 +1,134 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import { join } from 'path';
+import { Duration, CustomResource, Arn, ArnFormat, Stack } from 'aws-cdk-lib';
+import { IRole, PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import { Runtime } from 'aws-cdk-lib/aws-lambda';
+import { RetentionDays } from 'aws-cdk-lib/aws-logs';
+import { Provider } from 'aws-cdk-lib/custom-resources';
+import { Construct } from 'constructs';
+import { StreamingIngestionProps } from './model';
+import { createWorkgroupAndNamespacePolicyForRedshift } from '../../analytics/common/utils';
+import { AssociateIAMRoleToRedshift, ExistingRedshiftServerlessProps, ProvisionedRedshiftProps } from '../../analytics/private/model';
+import { createLambdaRole } from '../../common/lambda';
+import { POWERTOOLS_ENVS } from '../../common/powertools';
+import { SolutionNodejsFunction } from '../../private/function';
+
+export interface StreamingIngestionAssociateRoleProps {
+  readonly projectId: string;
+  readonly appIds: string;
+  readonly serverlessRedshift?: ExistingRedshiftServerlessProps;
+  readonly provisionedRedshift?: ProvisionedRedshiftProps;
+  readonly databaseName: string;
+  readonly dataAPIRole: IRole;
+  readonly streamingIngestionRole: IRole;
+  readonly streamingIngestionProps: StreamingIngestionProps;
+  readonly associateRoleTimeout: number;
+}
+
+export class StreamingIngestionAssociateRole extends Construct {
```

Comment:
Binding the IAM role to redshift should be a generic need. Let's make it a high-level construct.

Reply:
Please check #743 if it fits the use case in streaming ingestion.